### PR TITLE
(#4891) - add async/await examples to the docs

### PR DESCRIPTION
**Pull request by nolanlawson**
I'm very careful to note that this is an experimental ES7 API. However, I think we can do a great service to the JS community by providing a [Rosetta stone](https://en.wikipedia.org/wiki/Rosetta_Stone) to show the difference between callbacks/promises/async functions.

Also, I find myself starting to use ES7 async/await more and more, especially when I use PouchDB. Frankly I don't really find any feature of ES6 very exciting (except maybe arrow functions), but async/await vastly improves my productivity, because I can't seem to write promise-based code without messing it up. :stuck_out_tongue: 

Note that promises are still the default in the documentation.

screenshot 2016-02-14 15 59 10
screenshot 2016-02-14 15 59 18

### CODE REVIEW

File: docs/_includes/api/overview.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {% include anchor.html edit="true" title="API overview" hash="overview" %}
```
Comment (nolanlawson) [inline docs/_includes/api/overview.html]: The API overview is where I talk about the differences between callbacks, promises, and async functions. Note that I broke it up into separate subsections.

**Comment by nickcolley** [comment]
This is great!

Are they calling it ES2016 or ES7?

**Comment by nolanlawson** [comment]
I've heard it called both. Confusing, I know. :P

**Comment by daleharvey** [comment]
This is awesome large :+1:

File: docs/_includes/api/overview.html [hunk 2]
```diff
@@ -28,7 +33,26 @@
...
+To use a custom promise implementation with PouchDB, you must redefine a global `Promise` object before loading PouchDB:
+
+{% highlight html %}
+<script>window.Promise = MyCustomPromiseLibrary;</script>
+<script src="path/to/pouchdb.js"></script>
+{% endhighlight %}
+
+### Async functions
+
+If you are using a transpiler like [Babel](http://babeljs.io/), you can enable [async functions](http://babeljs.io/docs/plugins/syntax-async-functions/), which are an experimental API tentatively slated for release in ES7 (ES2016). This allows you to use the `async`/`await` keywords when consuming promise-based APIs like PouchDB's.
```
Comment (marten-de-vries) [inline docs/_includes/api/overview.html]: Actually, that plugin only parses async functions, it doesn't convert them. You need https://babeljs.io/docs/plugins/transform-async-to-generator/ for that. (Or my plugin [kneden](https://github.com/marten-de-vries/kneden), but that's not production ready yet...)

**Comment by marten-de-vries** [comment]
Apart from the minor nit above, this looks great :)

**Comment by nolanlawson** [comment]
I'll fix that suggestion and then merge. Thanks everyone!

**Comment by nolanlawson** [comment]
3acb34dad1fd24206a6e3dbe318ff355f91e6e8f

**Comment by nickcolley** [comment]
@nolanlawson thanks!

**Pull request closed without merging**